web-ui/src/components/notes/Note.jsx

```diff
@@ -1,77 +1,90 @@
-import React, { useContext, useEffect, useRef, useState } from "react";
-import { getNoteByCheckinId, updateCheckinNote } from "../../api/checkins";
-import useDebounce from "../../hooks/useDebounce";
+import React, { useContext, useEffect, useState } from "react";
+import {
+  getNoteByCheckinId,
+  createCheckinNote,
+  updateCheckinNote,
+} from "../../api/checkins";
+import { debounce } from "lodash/function";
 import NotesIcon from "@material-ui/icons/Notes";
 import LockIcon from "@material-ui/icons/Lock";
 import { AppContext } from "../../context/AppContext";
+import Snackbar from "@material-ui/core/Snackbar";
+import MuiAlert from "@material-ui/lab/Alert";
 
 import "./Note.css";
 
+async function realUpdate(note) {
+  let res = await updateCheckinNote(note);
+  if (res.error) {
+    console.error(res.error);
+  }
+}
+
+const updateNote = debounce(realUpdate, 1000);
+
+function Alert(props) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
+
 const Notes = (props) => {
   const { state } = useContext(AppContext);
   const { userProfile } = state;
   const canViewPrivateNote =
-    userProfile.role.includes("PDL") || userProfile.role.includes("ADMIN");
+    userProfile.memberProfile &&
+    userProfile.memberProfile.role &&
+    (userProfile.memberProfile.role.includes("PDL") ||
```

Comment:
This is insufficient. Hopefully the backend will save us, but the private note section shouldn't show up if I'm a PDL and I'm looking at my own check-in. With just this check, it would.